x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/session/IndexResolverFieldNamesTests.java

```diff
@@ -1316,6 +1373,18 @@ public void testCountAllWithEval() {
             """, Set.of("languages", "languages.*", "salary", "salary.*"));
     }
 
+    public void testCountAllWithEval_AndInlinestats() {
+        assertFieldNames("""
+            from test
+            | rename languages as l
+            | inlinestats max(salary) by l
+            | stats min = min(salary) by l
+            | eval x = min + 1
+            | stats ca = count(*), cx = count(x) by l
```

Comment:
I think the same behavior is expected when this `stats` is replaced by a `keep x, l` (no wildcard), right?

Maybe let's add such tests, and also some where the `STATS` or `KEEP` (no wildcard) comes before the `INLINESTATS`, for good measure.